tcbuilder/backend/bundle.py

```diff
@@ -577,14 +576,11 @@ def download_containers_by_compose_file(
                 # Use high-level API (no progress info).
                 image = dind_client.images.pull(image_name, platform=platform)
 
-            service['image'] = image.attrs['RepoDigests'][0]
+            svc_spec['image'] = image.attrs['RepoDigests'][0]
 
         log.info("Saving Docker Compose file")
         with open(os.path.join(manager.output_dir, "docker-compose.yml"), "w") as file:
-            # Serialization need to escape dollar sign and requires no
-            # environment variables interpolation.
-            file.write(compose.config.serialize.serialize_config(
-                config, escape_dollar=False).replace("@@MACHINE@@", "$MACHINE", 1))
```

Comment:
I imagine this was done as a hack to deal with the variables expansion before the "workaround" of using a double `$` was known. So once we merge this changes I imagine some internal pipeline will break.

This seems to be used in the bullseye branch of project `torizon-core-docker-evaluation` at least.

Reply:
Checking the commit history:

```
bundle: replace @@MACHINE@@ with $MACHINE when parsing docker-compose file

Replace @@MACHINE@@ with $MACHINE with $MACHINE when bundle command is parsing docker-compose file.

Related-to: TOR-1458
Signed-off-by: Dejan Jesenko <dejan.jesenko@toradex.com>
```

So it's probably worth checking ticket TOR-1458.

Reply:
Looking at TOR-1458 (now TCB-203), this workaround was made in order to avoid `${MACHINE}` from being expanded in the `command` field of the `portainer` service of the docker compose YAML used in the Portainer version of `torizon-core-docker-evaluation`:

```
command: --templates https://feeds.toradex.com/portainer/5/@@MACHINE@@/templates.json --logo https://docs.toradex.com/109812-torizon-logo-portainer.svg -H unix:///var/run/docker.sock
```